src/main/java/net/fabricmc/installer/Main.java

```diff
@@ -43,18 +40,23 @@
 
 	public static void main(String[] args) throws IOException {
 		String[] versionSplit = System.getProperty("java.version").split("\\.");
+		boolean showGui = args.length == 0 || !args[0].equals("nogui");
 		if (versionSplit.length == 2) { //Only check 1.x versions of java, new versions are formatted liked 12
 			int javaVersionMajor = Integer.parseInt(versionSplit[0]);
 			int javaVersionMinor = Integer.parseInt(versionSplit[1]);
 			if (javaVersionMinor < 8 && javaVersionMajor <= 1) {
 				System.out.println("You are using an outdated version of Java, Fabric will not work! Please update to Java 8 or newer to use Fabric.");
-				if (args.length == 0 || !args[0].equals("nogui")) {
+				if (showGui) {
 					JOptionPane.showMessageDialog(null, "You are using an outdated version of Java, Fabric will not work! Please update to Java 8 or newer to use Fabric.", "Java Version Warning", JOptionPane.ERROR_MESSAGE);
 				}
 				return;
 			}
 		}
-
+		try {
+			Utils.launcherCheck(showGui);
+		} catch (IOException e) {
+			e.printStackTrace();
```

Comment:
Maybe show a screen asking users to make sure minecraft launcher is closed if our fallback commands failed. (of course check showGui first)

Reply:
Is a message box ok? Or should it be a confirmation dialog?

Reply:
Hmm, do you think we should ask user to check about launcher status if we cannot obtain the process information? It's up to you how you handle this case. But just know that check may fail due to process info being inaccessible.